hi-ml-histopathology/testhisto/testhisto/utils/test_report_utils.py

```diff
@@ -10,7 +10,76 @@
 
 from health_azure.utils import download_file_if_necessary
 from histopathology.utils.report_utils import (collect_crossval_metrics, collect_crossval_outputs,
-                                               get_best_epoch_metrics, get_best_epochs, get_crossval_metrics_table)
+                                               crossval_runs_have_val_and_test_outputs, get_best_epoch_metrics,
+                                               get_best_epochs, get_crossval_metrics_table,
+                                               run_has_val_and_test_outputs)
+
+
+def test_run_has_val_and_test_outputs() -> None:
+    outputs_folder = "outputs/"
+    outputs_filename = "test_output.csv"
+    legacy_outputs_filename = outputs_folder + outputs_filename
+    val_outputs_filename = outputs_folder + "val/" + outputs_filename
+    test_outputs_filename = outputs_folder + "test/" + outputs_filename
+    arbitrary_filename = outputs_filename + "some_other_file"
+
+    run = MagicMock(display_name="mock run", id="abc123")
+
+    with patch.object(run, 'get_file_names', return_value=[legacy_outputs_filename]):
+        assert not run_has_val_and_test_outputs(run)
+
+    with patch.object(run, 'get_file_names', return_value=[val_outputs_filename, test_outputs_filename]):
+        assert run_has_val_and_test_outputs(run)
+
+    with patch.object(run, 'get_file_names', return_value=[val_outputs_filename]):
+        with pytest.raises(ValueError, match="does not have the expected files"):
+            run_has_val_and_test_outputs(run)
+
+    with patch.object(run, 'get_file_names', return_value=[test_outputs_filename]):
+        with pytest.raises(ValueError, match="does not have the expected files"):
+            run_has_val_and_test_outputs(run)
+
+    with patch.object(run, 'get_file_names', return_value=[arbitrary_filename]):
+        with pytest.raises(ValueError, match="does not have the expected files"):
+            run_has_val_and_test_outputs(run)
+
+
+def test_crossval_runs_have_val_and_test_outputs() -> None:
```

Comment:
[Same comments about hard-coded variables and loops for tests]

Reply:
Here each test case is a bit different, so I don't think a loop would be much cleaner...